string.c

```diff
@@ -4790,16 +4790,21 @@ static VALUE
 str_casecmp_p(VALUE str1, VALUE str2)
 {
     rb_encoding *enc;
-    VALUE folded_str1, folded_str2;
     VALUE fold_opt = sym_fold;
 
     enc = rb_enc_compatible(str1, str2);
     if (!enc) {
         return Qnil;
     }
 
-    folded_str1 = rb_str_downcase(1, &fold_opt, str1);
-    folded_str2 = rb_str_downcase(1, &fold_opt, str2);
+    if (ENC_CODERANGE(str1) == ENC_CODERANGE_7BIT &&
+        ENC_CODERANGE(str2) == ENC_CODERANGE_7BIT) {
+        VALUE cmp = str_casecmp(str1, str2);
+        return RBOOL(cmp == INT2FIX(0));
```

Comment:
Sorry, the previous benchmark would be my mistake.
When I tried again, I got the same result as you.
After measures it more finely, the breakeven point seems between `36*3` and `36*4`.

```suggestion
        static const long break_even_point = 120;
        long len1 = RSTRING_LEN(str1);
        if (len1 != RSTRING_LEN(str2)) return Qfalse;
        if (len1 < break_even_point) {
            VALUE cmp = str_casecmp(str1, str2);
            return RBOOL(cmp == INT2FIX(0));
        }
```